### ISSUE TITLE
--stream-url switch bug

### ISSUE THREAD

**Issue by johnthecracker** [streamlink/streamlink#559]
### Checklist

- [x] This is a bug report.
- [ ] This is a plugin request.
- [ ] This is a feature request.
- [ ] I used the search function to find already opened/closed issues or pull requests.

### Description
```
streamlink -h ( 0.3.1.814 )
  
  --stream-url
    If possible, translate the stream to a URL and print it.
```
_________________________
```
set url=http://www.filmon.com/channel/bbc-news
streamlink --stream-url %url% best
pause


error: The stream specified cannot be translated to a URL
Press any key to continue . . .
```
_________________________
```
set url=http://www.filmon.com/channel/bbc-news
streamlink --stream-priority "rtmp,hls,hds,http,akamaihd" %url% best 


[cli][info] Found matching plugin filmon for URL http://www.filmon.com/channel/bbc-news
error: No streams found on this URL: http://www.filmon.com/channel/bbc-news
Press any key to continue . . .
```

**Comment by beardypig** [comment]
This will be fixed with #560. However, it should be noted that not all filmon streams will produce a stream URL as they expire almost immediately.

**Comment by gravyboat** [comment]
I'm going to close this since we've merged #560, thanks for the report @johnthecracker!
